core/src/main/java/io/grpc/internal/InternalSubchannel.java

```diff
@@ -380,9 +383,11 @@ public void run() {
               addressIndex.reset();
               gotoNonErrorState(IDLE);
             } else {
-              pendingTransport.shutdown(
-                  Status.UNAVAILABLE.withDescription(
-                    "InternalSubchannel closed pending transport due to address change"));
+              if (pendingTransport != null) {
```

Comment:
`state.getState() == READY || CONNECTING` implies `pendingTransport != null`.

Reply:
> sets pendingTranport = null (line 632) then if at end calls index.reset() and schedules a backoff

And scheduling the backoff changes the state to TF (or IDLE):
https://github.com/grpc/grpc-java/blob/36e29abf4168646b4e3866061c00b5cbea6fcead/core/src/main/java/io/grpc/internal/InternalSubchannel.java#L296

Reply:
You're right.  Removed it.